library/src/main/java/pl/rspective/voucherify/android/view/VoucherCheckoutView.java

```diff
@@ -66,38 +85,75 @@ public void onClick(View v) {
             }
         });
 
-        // Load attributes
-        final TypedArray a = getContext().obtainStyledAttributes(attrs, R.styleable.VoucherCheckoutView);
+        invalidAnimation = AnimationUtils.loadAnimation(context, R.anim.invalid);
+        validAnimation = AnimationUtils.loadAnimation(context, R.anim.valid);
+        applyCustomAttributes(context, attrs, defStyle);
+    }
+
+    private void applyCustomAttributes(Context context, AttributeSet attrs, int defStyle) {
+        final TypedArray a = context.obtainStyledAttributes(attrs, R.styleable.VoucherCheckoutView, defStyle, 0);
 
-        validateButtonText = a.getString(R.styleable.VoucherCheckoutView_validateButtonText) ;
+        String voucherCodeHint = a.getString(R.styleable.VoucherCheckoutView_voucherCodeHint);
+        if (voucherCodeHint != null) {
+            voucherCodeLabel.setHint(voucherCodeHint);
+        }
+
+        String validateButtonText = a.getString(R.styleable.VoucherCheckoutView_validateButtonText);
         if (validateButtonText != null) {
             validateButton.setText(validateButtonText);
         }
 
+        voucherIcon = a.getDrawable(R.styleable.VoucherCheckoutView_voucherIcon);
+        if (voucherIcon == null) {
+            voucherIcon = ContextCompat.getDrawable(context, R.drawable.ic_voucherify_logo_white_24dp);
```

Comment:
Please extract this code into method and apply for two validVoucherIcon and invalidVoucherIcon as well.